light/stateprovider.go

```diff
@@ -451,26 +444,20 @@ func (s *StateProviderP2P) consensusParams(ctx context.Context, height int64) (t
 	var iterCount int64
 	for {
 		iterCount++
-		sig, err := retryAll()
+		err := retryAll()
 		if err != nil {
 			return types.ConsensusParams{}, err
 		}
+		// jitter+backoff the retry loop
+		timer.Reset(time.Duration(iterCount)*consensusParamsResponseTimeout +
```

Comment:
I think the underlying `retryAll` has its own retry loop, with its own timeout that causes a continue.  Now that this does not wait for retryAll to finish, would this create overlapping attempts?  (I wonder if it should cancel the `retryAll`? or maybe the underlying `retryAll` should not loop/retry?)

Reply:
Yea, re overlapping retries, added in a `cancel()` once any goroutine returns a valid response. Testing this on a node

Reply:
I believe this would block below on `<- timer.C` since that will wait for the timer that is reset to return to the channel. So based on my understanding it SHOULD wait to start another retryAll? cc @Kbhat1

Reply:
It does wait for timer start another `retryAll` but the previous `retryAll` keeps looping with its own retry loop.

Reply:
I think @stevenlanders may be right, the outer loop can still fire another `retryAll` before the inner goroutines have finished all their retries. Investigating this, think we can just remove the inner retries all together and just handle retries from the outer loop. cc: @udpatil

Reply:
@udpatil @stevenlanders removed inner retry + added in a child context

Reply:
lgtm